src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -240,42 +240,59 @@ internal partial class CodeFixService : ICodeFixService
             }
         }
 
-        public async Task<CodeFixCollection?> GetDocumentFixAllForIdInSpanAsync(
+        public Task<CodeFixCollection?> GetDocumentFixAllForIdInSpanAsync(
             Document document, TextSpan range, string diagnosticId, CodeActionOptions options, CancellationToken cancellationToken)
+            => GetDocumentFixAllForIdInSpanAsync(document, range, diagnosticId, DiagnosticSeverity.Hidden, options, cancellationToken);
+
+        public async Task<CodeFixCollection?> GetDocumentFixAllForIdInSpanAsync(
+            Document document, TextSpan range, string diagnosticId, DiagnosticSeverity severity, CodeActionOptions options, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            var diagnostics = (await _diagnosticService.GetDiagnosticsForSpanAsync(document, range, diagnosticId, includeSuppressedDiagnostics: false, cancellationToken: cancellationToken).ConfigureAwait(false)).ToList();
-            if (diagnostics.Count == 0)
+            var diagnostics = (await _diagnosticService.GetDiagnosticsForSpanAsync(document, range, diagnosticId, includeSuppressedDiagnostics: false, cancellationToken: cancellationToken).ConfigureAwait(false));
+            diagnostics = diagnostics.WhereAsArray(d => d.Severity.GreaterThanOrEqualTo(severity));
+            if (!diagnostics.Any())
                 return null;
 
             using var resultDisposer = ArrayBuilder<CodeFixCollection>.GetInstance(out var result);
             var spanToDiagnostics = new SortedDictionary<TextSpan, List<DiagnosticData>>
```

Comment:
why a List and not an ImmutableArray?

Reply:
`StreamFixesAsync` takes an argument of type `SortedDictionary<TextSpan, List<DiagnosticData>>` here https://github.com/dotnet/roslyn/blob/fab4fb3d84789a06e02c7f60eba32f0f6c87adaa/src/Features/Core/Portable/CodeFixes/CodeFixService.cs#L358

@CyrusNajmabadi I believe you wrote this code, do you recall why you used `List<DiagnosticData>`? Happy to change it to `ImmutableArray<<DiagnosticData>>`